lib/assert.js

```diff
@@ -47,8 +47,10 @@ class AssertionError extends Error {
 		// Reserved for power-assert statements
 		this.statements = [];
 
-		if (opts.stack) {
+		if (typeof opts.stack === 'string') {
 			this.stack = opts.stack;
+		} else if (opts.stack instanceof Error) {
+			this.stack = opts.stack.stack;
```

Comment:
It may make sense to rename `opts.stack`. Perhaps `opts.objectWithStack`? Same for the `stack` variable and `getStack()` function.

Reply:
Agreed.